qemu/tests/virtio_mem_dynamic_memslots_with_migration.py

```diff
@@ -0,0 +1,54 @@
+import time
+
+from virttest import error_context
+
+from virttest.utils_misc import normalize_data_size
+from provider import virtio_mem_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Virtio-mem dynamic memslots with migration test
+    1) Boot VM
+    2) Resize virtio-mem device
+    3) Do migration
+    4) Check virtio-mem size in dst host
+    5) Validate the virtio-mem device now has the correct number of memslots
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    requested_size_vmem_test = params.get("requested-size_test_vmem0")
+    total_memslots = params.get_numeric("total_memslots", 4)
+    mem_object_id = params.get("mem_devs", "vmem0")
+    timeout = params.get_numeric("timeout", 10)
+
+    virtio_mem_model = "virtio-mem-pci"
+    if "-mmio:" in params.get("machine_type"):
+        virtio_mem_model = "virtio-mem-device"
+    vmem_dev = vm.devices.get_by_params({"driver": virtio_mem_model})[0]
+    device_id = vmem_dev.get_qid()
```

Comment:
From my perspective, it is not a good design to add the logic of getting `virtio_mem_model`  for different platform in test case level, which is coupling with the logic in avocado-vt.   In other word,  all individual cases in tp-qemu need be update if there is any product change,  instead of one update in avocado-vt.

The parameter `vm_memdev_model` is an indicator but it is not shown in the vm object with the current design for `memory_define_by_params`,  it need be improved to support this usage.

If considering a workaround without change the avocado-vt,  take one step back,  I think use the coupling device id naming rule[1] with avocado-vt is more stable than using the logic in L32~L34.   The device id is automation stuff, nothing related with product.  In other word, i prefer to use the device id: `virtio_mem-$mem_dev` directly.

cc @PaulYuuu as we have some discussion offline ^^


[1] https://github.com/avocado-framework/avocado-vt/blob/a434e270634e81a40364a06c9085f7b6918ba2c3/virttest/qemu_devices/qcontainer.py#L3532

Reply:
Thanks for the suggestion @yanan-fu, code updated !